### PR TITLE
Check changelog extensions

### PR DESCRIPTION
**Pull request by daverodgman**
## Description

Catch changelog entries that lack a .txt extension (this seems to happen regularly with each release).

## PR checklist

Please tick as appropriate and edit the reasons (e.g.: "backport: not needed because this is a new feature")

- [x] **changelog** not required - not user visible
- [x] **backport** #8293 
- [x] **tests** not required

### CODE REVIEW

**Comment by gilles-peskine-arm** [comment]
We do need a backport since the same risk exists in all maintained branches.

File: scripts/assemble_changelog.py [hunk 1]
```diff
@@ -448,14 +453,23 @@ def list_files_to_merge(options):
     files_to_merge.sort(key=EntryFileSortKey)
     return files_to_merge
 
+def check_extensions(options):
+    files = glob.glob(os.path.join(options.dir, '*'))
+    files = {x for x in files if not x.endswith(".txt")}
+    files.discard("ChangeLog.d/00README.md")
```
Comment (gilles-peskine-arm) [inline scripts/assemble_changelog.py:459]: Skipping `README.md` this way doesn't work with a different `options.dir` or on Windows.
```suggestion
    files = {x for x in files if not x.endswith(".txt") and not os.path.basename(x) == '00README.md'}
```

File: scripts/assemble_changelog.py [hunk 2]
```diff
@@ -448,14 +453,23 @@ def list_files_to_merge(options):
...
 def merge_entries(options):
     """Merge changelog entries into the changelog file.
 
     Read the changelog file from options.input.
+    Check that all entries have a .txt extension
     Read entries to merge from the directory options.dir.
     Write the new changelog to options.output.
     Remove the merged entries if options.keep_entries is false.
     """
+    check_extensions(options)
```
Comment (gilles-peskine-arm) [inline scripts/assemble_changelog.py:472]: The implementation of this check is closely tied to the implementation of `list_files_to_merge`, so `check_extensions` should be called from (or even inlined into) `list_files_to_merge`.

**Review by gilles-peskine-arm — CHANGES_REQUESTED** [review]
I agree with the objective, not with the implementation.

File: scripts/assemble_changelog.py (hunk 2 above)
Comment (minosgalanakis) [reply scripts/assemble_changelog.py:472]: Or could we just make `list_files_to_merge` throw the newly introduced exception?

**Review by gilles-peskine-arm — APPROVED** [review]
LGTM

**Review by tom-cosgrove-arm — APPROVED** [review]
LGTM

**Comment by gilles-peskine-arm** [comment]
You need to rename the existing bad files for the CI to pass.

**Comment by daverodgman** [comment]
Waiting for #8292 before this can be merged, or CI will start failing. CI otherwise green so considering this a CI pass.

**Review by gilles-peskine-arm — APPROVED** [review]

**Pull request merged**